.github/workflows/rust_checks.yml

```diff
@@ -8,80 +8,63 @@ on:
 env:
   CARGO_TERM_COLOR: always
 
-jobs:
+jobs: 
   check:
     name: Minimal checks on non-zkvm crates
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Download and Extract Fixtures
-        run: |
-          chmod +x ./scripts/download-and-extract-fixtures.sh
-          ./scripts/download-and-extract-fixtures.sh
-
       - name: Install Rust toolchain
         uses: dtolnay/rust-toolchain@nightly
         with:
           components: clippy, rustfmt
-
-      - name: Cache Rust dependencies
-        uses: actions/cache@v4
+      
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
+          key: base 
 
       - name: Check formatting
         run: cargo fmt --check --all
 
       - name: Run tests
         run: cargo test -p zkevm-metrics -p witness-generator -p ere-hosts
 
+  download-eest-fixtures:
+    name: Download and Extract Fixtures
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Download and Extract Fixtures
+        run: |
+          chmod +x ./scripts/download-and-extract-fixtures.sh
+          ./scripts/download-and-extract-fixtures.sh
+
   check-ere-host:
     name: Checks crate ere-hosts
     strategy:
+      fail-fast: false
       matrix:
-        include:
-          - zkvm: sp1
-          - zkvm: risc0
-          - zkvm: openvm
-          - zkvm: pico
-          - zkvm: zisk
+        zkvm: [sp1, risc0, openvm, pico, zisk]
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Download and Extract Fixtures
-        run: |
-          chmod +x ./scripts/download-and-extract-fixtures.sh
-          ./scripts/download-and-extract-fixtures.sh
-
       - name: Install Rust toolchain
         uses: dtolnay/rust-toolchain@nightly
         with:
           components: clippy, rustfmt
 
-      - name: Cache Rust dependencies
-        uses: actions/cache@v4
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
+          key: ${{ matrix.zkvm }}
```

Comment:
Include the zkVM as part of the cache key, so different zkVMs don't clash using the same cache since cargo would invalidate the whole cache since detected compilation flags changes which was very unfortunate.

Now each zkVM has its own cache thus subsequent runs can leverage it correctly without another zkVM destroying it.